sdk/maps/azure-maps-render/azure/maps/render/_render_client.py

```diff
@@ -441,12 +441,12 @@ def get_copyright_from_bounding_box(
 
         _include_text=kwargs.pop("include_text", True)
 
-        return self._render_client.get_copyright_from_bounding_box(
-            south_west=(bounding_box.south,bounding_box.west),
-            north_east=(bounding_box.north,bounding_box.east),
+        return Copyright(**self._render_client.get_copyright_from_bounding_box(
+            south_west=[bounding_box.south,bounding_box.west],
+            north_east=[bounding_box.north,bounding_box.east],
             include_text= "yes" if _include_text else "no",
             **kwargs
-        )
+        ).__dict__)
```

Comment:
What was the mypy error here?

Reply:
Expected return type is from: `sdk/maps/azure-maps-render/azure/maps/render/models/_models.py`
But the one `self._render_client.get_copyright_from_bounding_box` is returning is from the _generated dir: `sdk/maps/azure-maps-render/azure/maps/render/_generated/models/_models.py`. We have to convert generated type to the custom one.

Reply:
Ah good point. However, it seems that the tests and samples for this library do expect `Copyright` in the shape of the generated model. I don't have context for whether this was intended to be transformed to the custom type (and overlooked), or if we decided the generated type was acceptable. Let me see if I can tag the library owner to weigh in.

Reply:
@alextts627 - can you help answer this question?

Reply:
That's the way these things are handled within search, route, geolocation.. But let's double check :)

One more thing... I need some help regarding autorest. I wasn't able to generate py3 files in models dir: sdk/maps/azure-maps-search/azure/maps/search/_generated/models/_models_py3.py

I was using this swagger file as input: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/maps/data-plane/Search/preview/1.0/search.json

Using legacy msrest mode I was able to generate just: _models.py, but not _models_py3.py.

Note: I fixed issue in the doc, by replacing: `["SharedKey"]["in"] = "header"`

Can you help me with concrete parameters for generating code.. This existing README.md is not up to date. I also tried with cloning azure-rest-api-specs repo and couldn't produce _models_py3.py file.

Thanks :)

Reply:
Just for clarity, the reason I want to double-check with the library owner on this is because from generated type --> custom type, it's unclear to me whether the intention there was to rename the `regions` property to `regional_copyrights`. That's not how things will deserialize today, so we'll need to update the code if that was the intention (and document the breaking change).

Regarding autorest - that's by design. We no longer generate with a _models_py3 file, that was only necessary during the time that the SDK still supported python 2.7 (py2 and py3 models used to be separated). I'm guessing it's been awhile since this library has been regenerated and so it still had the legacy _models_py3 file.

Why do you believe the existing readme to not be up to date? From what I can tell, the tag in the README matches the default API version for this version of the library.

Reply:
@kristapratico thanks a lot for explanation

Ok, now I understand why I couldn't generate _models_py3.py file. But, hm.. reverse search uses models from _models_py3.py to validate response (which causes this error by the way: https://github.com/Azure/azure-sdk-for-python/issues/32565.. I created pr for that which we can merge as this py3 is still used, but not generated with autorest: https://github.com/Azure/azure-sdk-for-python/pull/34977)? 

The other thing is why README.md seems outdated to me is that I couldn't generate _models.py (regular not py3) without using msrest mode (which is legacy and it is removed: https://github.com/Azure/azure-rest-api-specs/pull/28269). Is there any other way to modify README.md for azure-maps-search to generate _models.py ?

Reply:
> Ok, now I understand why I couldn't generate _models_py3.py file. But, hm.. reverse search uses models from _models_py3.py to validate response (which causes this error by the way: https://github.com/Azure/azure-sdk-for-python/issues/32565.. I created pr for that which we can merge as this py3 is still used, but not generated with autorest: https://github.com/Azure/azure-sdk-for-python/pull/34977)?

In this case, the spec needs to be updated to match the service response. The py3 file will be deleted upon next generation so the update won't persist if we change it manually. Additionally, since the spec file is used to generate more languages than just Python, fixing this in the spec will propagate to all the other libraries.

> The other thing is why README.md seems outdated to me is that I couldn't generate _models.py (regular not py3) without using msrest mode (which is legacy and it is removed: https://github.com/Azure/azure-rest-api-specs/pull/28269). Is there any other way to modify README.md for azure-maps-search to generate _models.py ?

Since this library uses swagger-based code generation, `msrest` mode has to be used if models are needed. The Python SDK is transitioning from generation from swagger definitions to [typespec](https://typespec.io/) and the service definition for maps has not be converted over yet. All new services / versions are defined in typespec, but some of the existing / not recently updated services haven't made the jump yet.

Reply:
@kristapratico Thanks, just regarding specs, it's already updated: https://github.com/Azure/azure-rest-api-specs/pull/17151

Reply:
> @kristapratico Thanks, just regarding specs, it's already updated: [Azure/azure-rest-api-specs#17151](https://github.com/Azure/azure-rest-api-specs/pull/17151)

Great, thanks for sharing! In that case, is it possible to update https://github.com/Azure/azure-sdk-for-python/pull/34977 with the latest code generation? For anywhere the code uses models_py3, can we update references to use _models.py?

Reply:
Upon trying to regenerate new code described in this draft pr: https://github.com/Azure/azure-sdk-for-python/pull/35029 I managed to resolve the issue (it will replace usage of _models_py3.py with regular _models.py), but there is a lot of work to be done before fixing tests/everything.. Can you please take a brief look at that :)

Once again, thanks a lot for your help! :D